http/src/main/scala/org/http4s/blaze/http/Http1ServerCodec.scala

```diff
@@ -295,7 +298,9 @@ private final class Http1ServerCodec(maxNonBodyBytes: Int, pipeline: TailStage[B
       private var prelude: StringBuilder,
       maxCacheSize: Int) extends InternalWriter {
 
-    prelude.append("transfer-encoding: chunked\r\n\r\n")
+    // The transfer-encoding header will be the last header, and before each write,
+    // we will append a '\r\n{length}\r\n
+    prelude.append("transfer-encoding: chunked\r\n")
```

Comment:
A number of headers, including `content-length`, are defined in a shared location and are used by http/2, so I'm going to leave it as is so that the case of the `content-length` header doesn't change depending on whether its added automatically by the writer or by another mechanism.